File: WebLab3/src/main/java/com/arslanka/backend/models/views/Triangle.java
package com.arslanka.backend.models.views;

public class Triangle implements Shape {
    private final Double k;
    private final Double bMulti;

    public Triangle(Distance xRad, Distance yRad) {
        if (xRad.equals(yRad)) {
            k = 1.0d;
        } else if (xRad == Distance.R) {
            k = 1.0d / 2.0d;
        } else {
            k = 2.0d;
        }

        if (yRad == Distance.R) {
            bMulti = 1.0d;
        } else {
            bMulti = 1.0d / 2.0d;
        }
    }

    @Override
    public Boolean isHit(Double radius, Double x, Double y) {
        double b = bMulti * radius;
        y = Math.abs(y);
        x = Math.abs(x);
        return y <= (-k * x + b);
    }
}

File: web-basics/WebLab2/src/main/kotlin/com/arslanka/weblab2/models/view/Triangle.kt
package com.arslanka.weblab2.models.view

import java.math.BigDecimal

class Triangle(xDistance: Distance, yDistance: Distance) : Shape() {
    private var k = BigDecimal.ZERO
    private var bMulti = BigDecimal.ZERO

    init {
        k = if (xDistance == yDistance) {
            BigDecimal.ONE
        } else if (xDistance === Distance.R) {
            BigDecimal.valueOf(0.5)
        } else {
            BigDecimal.valueOf(2)
        }
        bMulti = if (yDistance === Distance.R) BigDecimal.ONE else BigDecimal.valueOf(0.5)
    }

    override fun isHit(radius: BigDecimal, x: BigDecimal, y: BigDecimal): Boolean {
        val b = bMulti * radius
        return y.abs() <= (-k * x.abs() + b)
    }
}